### ISSUE TITLE
Add an IterationLitstener that shows samples per second 

### ISSUE THREAD

**Issue by wmeddie** [deeplearning4j/deeplearning4j#1985]
Would also be nice if it output the time since it last printed like so:

    o.d.o.l.SpeedIterationListener - iteration 1000: loss = 2.521898223350428; time = 2.3980s; samples/sec = 13344.4

**Comment by raver119** [comment]
Implemented: https://github.com/deeplearning4j/deeplearning4j/pull/1987

Will be merged later, though.

**Comment by raver119** [comment]
P.s. Here's how it looks like. All output fields are optional.

```
o.d.o.l.PerformanceListener - iteration 4; iteration time: 360 ms; samples/sec: 355.556; batches/sec: 2.778; score: 2.059811123111661;
```
